Tools/refcounts/lint.py

```diff
@@ -0,0 +1,283 @@
+        else:
+            # process parameter
+            if name is None:
+                w.error(lineno, f'missing parameter name in {line!r}')
+                continue
+            sig: Signature = signatures[func]
+            if name in sig.params:
+                w.error(lineno, f'duplicated parameter name in {line!r}')
+                continue
+            sig.params[name] = Param(name, lineno, ctype, reftype, comment)
+
+    if w.count:
+        print()
+        print(f"Found {w.count} issue(s)")
+
+    return FileView(signatures, frozenset(incomplete))
+
+class CheckerWarnings:
+    def __init__(self) -> None:
+        self.count = 0
+
+    def block(self, sig: Signature, message: str) -> None:
+        self.count += 1
+        print(f'{flno_(sig.lineno)} {sig.name:50} {message}')
+
+    def param(self, sig: Signature, param: Param, message: str) -> None:
+        self.count += 1
+        fullname = f'{sig.name}[{param.name}]'
+        print(f'{flno_(param.lineno)} {fullname:50} {message}')
+
+def check(view: FileView) -> None:
+    w = CheckerWarnings()
+
+    for sig in view.signatures.values():  # type: Signature
+        # check the return value
+        rparam = sig.rparam
+        if not rparam.ctype:
+            w.block(sig, "missing return value type")
+        if rparam.reftype is RefType.UNKNOWN:
+            w.block(sig, "unknown return value type")
+        # check the parameters
+        for name, param in sig.params.items():  # type: (str, Param)
+            ctype, reftype = param.ctype, param.reftype
+            if ctype in OBJECT_TYPES and reftype is RefType.UNUSED:
+                w.param(sig, param, "missing reference count management")
+            if ctype not in OBJECT_TYPES and reftype is not RefType.UNUSED:
+                w.param(sig, param, "unused reference count management")
+            if name != C_ELLIPSIS and not name.isidentifier():
+                # Python accepts the same identifiers as in C
+                w.param(sig, param, "invalid parameter name")
+
+    if w.count:
+        print()
+        print(f"Found {w.count} issue(s)")
+    names = view.signatures.keys()
+    if sorted(names) != list(names):
+        print("Entries are not sorted")
+
+def check_structure(view: FileView, stable_abi_file: str) -> None:
+    stable_abi_str = Path(stable_abi_file).read_text()
+    stable_abi = tomllib.loads(stable_abi_str)
+    expect = stable_abi['function'].keys()
+    # check if there are missing entries (those marked as "TODO" are ignored)
+    actual = IGNORE_LIST | view.incomplete | view.signatures.keys()
+    if missing := (expect - actual):
+        print('[!] missing stable ABI entries:')
+        for name in sorted(missing):
+            print(name)
+
+def _create_parser() -> ArgumentParser:
+    parser = ArgumentParser(prog='lint.py')
+    parser.add_argument('file', help="the file to check")
+    parser.add_argument('--stable-abi', help="the stable ABI TOML file to use")
```

Comment:
Can we make this argument optional and bind it to a default parameter, like we did in `Tools/cases_generator/...`?

Reply:
Sure!

Reply:
(Sorry for the force push, but I wanted to have a clean initial commit)

Reply:
The tool can be used without any argument. It can use the default ABI file from `Misc` if you just specify `--abi` or a specific ABI file if you use `--abi=FILE`.